test/fixtures/contract_hash.json

```diff
@@ -10,7 +10,7 @@
       "ticker": "ACE",
       "version": 0
     },
-    "contractHash": "2d9d310ab7885463357382856ec91c7f586b4d2dabfac23849c66aef3d378310"
+    "contractHash": "1083373def6ac64938c2faab2d4d6b587f1cc96e85827335635488b70a319d2d"
```

Comment:
looks like blocktream explorer is using little endian encoding: https://blockstream.info/liquid/asset/0c5d451941f37b801d04c46920f2bc5bbd3986e5f56cb56c6b17bedc655e9fc6

now I'm doubting :thinking:

Reply:
It should *display* in little-endian, but the hash should be using big-endian when the entropy is calculated.

In issue https://github.com/Blockstream/asset_registry/issues/51, the referenced asset id is a9d52b02348a097d4f780644064c437e364a41d6299290fef1c49e7d1fbbf665

This is the code he used to issue the asset: https://github.com/baset-btc/liquid-test/blob/main/test.js

If we look at the asset using [esplora](https://blockstream.info/liquidtestnet/asset/a9d52b02348a097d4f780644064c437e364a41d6299290fef1c49e7d1fbbf665) we'll see the contract_hash is ed90ec2399bbcca8b38378a1bef6db81127a749ba84b791bacd028183a5faa95. This is the big-endian value of the contract hash, which can be verified by changing the values of [this script](https://github.com/Randy808/btcplusplusdemo/blob/main/generate_contract_hash.py) (you can delete everything after contract_hash = hashlib.sha256(contract_json.encode("ascii")).hexdigest() and just print out the contract_hash). It happens because the little-endian value was used to generate the entropy and esplora is expecting to have to convert it.

In the elements code, [ParseHashV performs length and hex validation](https://github.com/ElementsProject/elements/blob/elements-23.2.1/src/rpc/util.cpp#L105). The string hex is then passed into uint256S(strHex). The [constructor for uint256S](https://github.com/ElementsProject/elements/blob/elements-23.2.1/src/uint256.h#L151) then calls [SetHex](https://github.com/ElementsProject/elements/blob/elements-23.2.1//src/uint256.cpp#L65) on a newly initialized uint256. SetHex converts the string to a char* and calls another [SetHex](https://github.com/ElementsProject/elements/blob/elements-23.2.1//src/uint256.cpp#L37) where the decoding happens. The variable digits [is set to the last valid hex character](https://github.com/ElementsProject/elements/blob/elements-23.2.1//src/uint256.cpp#L51) and the digits [are read in reverse order into p1](https://github.com/ElementsProject/elements/blob/elements-23.2.1//src/uint256.cpp#L55-L60) which points to the start of m_data (the data of the uint256)

Reply:
ok right :+1: 

however, I would revert the changes to the fixtures because they are coming from esplora API (let's keep the string encoded hash reversed). Instead, could you reverse the contractHash in the test file?

Reply:
Yeah, will do 👍